app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminSetCustomerActiveViaGridActionGroup.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminSetCustomerActiveViaGridActionGroup">
+        <annotations>
+            <description>Sets a Customer to the Active status from the admin All Customers grid</description>
+        </annotations>
+        <arguments>
+            <argument name="customerEmail" type="string" defaultValue="{{Simple_US_CA_Customer.email}}"/>
+        </arguments>
+        <waitForElementVisible selector="{{AdminCustomerGridMainActionsSection.customerCheckbox(customerEmail)}}" stepKey="waitForCustomerRow"/>
+        <click selector="{{AdminCustomerGridMainActionsSection.customerCheckbox(customerEmail)}}" stepKey="selectCustomer"/>
+        <click selector="{{AdminCustomerGridMainActionsSection.actions}}" stepKey="clickActions"/>
+        <waitForElementVisible selector="{{AdminCustomerGridMainActionsSection.setActive}}" stepKey="waitForDropDownOpen"/>
+        <click selector="{{AdminCustomerGridMainActionsSection.setActive}}" stepKey="clickSetActive"/>
+        <waitForPageLoad stepKey="waitForLoad"/>
+        <waitForText selector="{{AdminMessagesSection.success}}" userInput="A total of 1 record(s) were updated." stepKey="seeSuccessMessage"/>
```

Comment:
Could you use existing `Assert` Action Group in Test, as in some cases we might like to assert that the action fails?

Reply:
@lbajsarowicz are you suggesting that i should remove the `waitForText` line from this actiongroup & use an assert actiongroup after i reference `AdminSetCustomerActiveViaGridActionGroup` in the test?

my assumption was that this actiongroup i wrote could be for the positive scenario & if someone needs to specifically test the negative case, then they could extend this actiongroup & merge with the `waitForText` action 🤷 

what are your thoughts between the two approaches?

Reply:
You can rename the Action Group then to include state assertion `Successfully`.

Reply:
done